packages/analytics/amplify_analytics_pinpoint_dart/test/analytics_legacy_native_data_test.dart

```diff
@@ -0,0 +1,120 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import 'dart:async';
+
+import 'package:amplify_analytics_pinpoint_dart/amplify_analytics_pinpoint_dart.dart';
+import 'package:amplify_analytics_pinpoint_dart/src/impl/analytics_client/endpoint_client/endpoint_store_keys.dart';
+import 'package:amplify_core/amplify_core.dart';
+import 'package:amplify_secure_storage_dart/amplify_secure_storage_dart.dart';
+import 'package:mocktail/mocktail.dart';
+import 'package:test/test.dart';
+
+void main() {
+  group('Analytics Legacy Native Data Tests', () {
+    late MockLegacyNativeDataProvider legacyDataProvider;
+    late MockSecureStorage store;
+
+    const appId = 'appId';
+    const legacyEndpointId = 'legacy-endpointId';
+
+    setUp(() {
+      legacyDataProvider = MockLegacyNativeDataProvider();
+      store = MockSecureStorage();
+    });
+
+    test('First app load, no legacy data, writes proper values', () async {
+      when(() => legacyDataProvider.getEndpointId(appId))
+          .thenAnswer((_) => Future.value());
```

Comment:
Explicitly show that null is being returned

```suggestion
          .thenAnswer((_) => Future.value(null));
```

Or better yet:

```suggestion
          .thenAnswer((_) async => null);
```

Reply:
Alright thanks!